### PR TITLE
improve toc in default template

### PR DESCRIPTION
**Pull request by superyyrrzz**
1. fix toc item with empty href getting parent toc's href by mustache fallback...
2. fix breadcrumb rendering when current page occurs more than once in toc.

@chenkennt @hellosnow @qinezh @vwxyzh @ansyral

### CODE REVIEW

File: src/docfx.website.themes/default/toc.html.js [hunk 1]
```diff
@@ -6,6 +6,11 @@ exports.transform = function (model) {
   return model;
 
   function transformItem(item, level) {
+    // set to null incase mustache looks up
+    item.href = item.href || null;
+    item.tocHref = item.tocHref || null;
```
Comment (hellosnow) [inline src/docfx.website.themes/default/toc.html.js]: >tocHref [](start = 9, length = 7)

not sure whether to update topicHref also?

Comment (hellosnow) [inline src/docfx.website.themes/default/toc.html.js]: topicHref is not used in js, so may not update?

---
In reply to: [86301821](https://github.com/dotnet/docfx/pull/853#discussion_r86301821) [](ancestors = 86301821)

**Comment by hellosnow** [comment]
:shipit:

File: src/docfx.website.themes/default/toc.html.js (hunk 1 above)
Comment (superyyrrzz) [inline src/docfx.website.themes/default/toc.html.js]: It's not used in mustache template, so no need to prevent fallback.

**Pull request merged**